### PR TITLE
Added docker setup and .husky

### PR DESCRIPTION
**Pull request by ArpitBlagan**
Issue: #14
https://www.loom.com/share/36b8e5a3b8724c5a901a10c445ad6a75?sid=198c0357-275d-4c54-a034-47e8597a8b5b

### CODE REVIEW

**Review by youngbloodcyb — CHANGES_REQUESTED** [review]
Love this, thanks for taking this on. 

Only thing is looks like prettier config is different and it's changing double quotes to single quotes throughout the whole codebase. Can you update that to match the current codebase?

**Comment by ArpitBlagan** [comment]
> Love this, thanks for taking this on.
> 
> Only thing is looks like prettier config is different and it's changing double quotes to single quotes throughout the whole codebase. Can you update that to match the current codebase? 

prettierrc is updated 
and check you twitter @youngbloodcyb 
thank you

**Comment by vtfg** [comment]
It would be great if you replaced all relative imports (../ui i.e.) with absolute imports (@/components/ui).

**Comment by ArpitBlagan** [comment]
> It would be great if you replaced all relative imports (../ui i.e.) with absolute imports (@/components/ui).

can you tell me where I am using relative imports that will be helpfull

**Comment by vtfg** [comment]
Not you, but there are some relative imports in some files. Since you are linting them all, you could padronize everything.

Search for `../` in VSCode and update them to use absolute imports, like this:

- `../components` => `@/components/`
- `../ui` => `@/components/ui`
- `../auth` => `@/components/auth`
- `../lib` => `@/lib`
- `../db` => `@/lib/db`

It's a code quality thing, it won't affect anything.

**Comment by ArpitBlagan** [comment]
> Not you, but there are some relative imports in some files. Since you are linting them all, you could padronize everything.
> 
> Search for `../` in VSCode and update them to use absolute imports, like this:
> 
> * `../components` => `@/components/`
> * `../ui` => `@/components/ui`
> * `../auth` => `@/components/auth`
> * `../lib` => `@/lib`
> * `../db` => `@/lib/db`

okay sorry i misunderstood

**Comment by ArpitBlagan** [comment]
> Not you, but there are some relative imports in some files. Since you are linting them all, you could padronize everything.
> 
> Search for `../` in VSCode and update them to use absolute imports, like this:
> 
> * `../components` => `@/components/`
> * `../ui` => `@/components/ui`
> * `../auth` => `@/components/auth`
> * `../lib` => `@/lib`
> * `../db` => `@/lib/db`
> 
> It's a code quality thing, it won't affect anything.

Done sir. 
I am currently looking for some unique project ideas to standout from the crowd if you have any please let me know @vtfg

**Comment by vtfg** [comment]
> I am currently looking for some unique project ideas to standout from the crowd

"Unique" is a strong keyword. I think it is very hard to have an unique idea, specially in today's world. What you can do is: find a project you like and re-develop it, improving or adding some features.

I'm not the best guy to ask this. I also have trouble finding project ideas.

File: package.json [hunk 1]
```diff
@@ -87,7 +94,7 @@
     "@vitejs/plugin-react": "^4.3.1",
     "autoprefixer": "^10.0.1",
     "drizzle-kit": "^0.24.2",
-    "eslint": "^8",
+    "eslint": "^8",=
```
Comment (vtfg) [inline package.json]: The `=` is causing an installation error.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      - db_data:/var/lib/postgresql/data
+
+  web:
+    build:
+      context: .
+      dockerfile: Dockerfile
+    ports:
+      - "3000:3000"
+    env_file:
+      - .env
```
Comment (vtfg) [inline docker-compose.yml]: The project uses the `.env.local` file.

I actually don't think that adding `env_file` here has any impact, as loading it is the Next.js/Drizzle's responsability.

> Edit: I tested and it really had no impact.

**Review by youngbloodcyb — CHANGES_REQUESTED** [review]
sorry to keep going back and forth on this. I'd really prefer if we kept each PR separate to its respective feature / bug. In this case since the purpose is to add capability for docker, lets focus on that. 

I think the only files that should be added / edited are the `.dockerignore`, `.husky/pre-commit` and `Dockerfile`. For adding docker there's no reason 100 files should be updated. 

Let's strip this PR down and only have the edits for those 3 files so that it's easier to review and test.

File: docker-compose.yml (hunk 2 above)
Comment (ArpitBlagan) [reply docker-compose.yml]: ooh yha let me change it to .env.local as I don't want to put the env file in the image.

**Comment by youngbloodcyb** [comment]
> Not you, but there are some relative imports in some files. Since you are linting them all, you could padronize everything.
> 
> Search for `../` in VSCode and update them to use absolute imports, like this:
> 
> * `../components` => `@/components/`
> * `../ui` => `@/components/ui`
> * `../auth` => `@/components/auth`
> * `../lib` => `@/lib`
> * `../db` => `@/lib/db`
> 
> It's a code quality thing, it won't affect anything.

I think this is a great idea but would rather have this be its own PR to test. @ArpitBlagan if you'd like, you could open this up separately from the docker init

**Comment by ArpitBlagan** [comment]
closing this one and creating a new separate pr's

**Pull request closed without merging**